torch/_inductor/ir.py

```diff
@@ -2755,9 +2750,12 @@ def fake_reindex(index):  # type: ignore[no-untyped-def]
             if unbacked_symbols_in_sizes and (not is_contiguous_storage_and_layout(x)):
                 # realize x; otherwise, the dynamic_reshape_indexer below will fail
                 # due to the size_hint's inability to process unbacked SymInts
-                x = ExternKernel.realize_input(x)
+                # TODO: unbacked should not diverge from backed in determining striding
+                # Need to require contiguous here instead of realize, see:
+                # https://github.com/pytorch/pytorch/issues/145561
+                x = ExternKernel.require_contiguous(x)
```

Comment:
This is fine following the existing logic, but I think `unbacked_symbols_in_sizes` doesn't necessarily mean `x` should be contiguous, so a better way is to realize x and create different view based on x's strides?

Reply:
This will create a contiguous copy if x is not contiguous or having symints? (e.g. an non contiguous ReinterpretView of a contiguous buffer)

Reply:
Initially, this branch is operating under the assumption that `x` is contiguous. Then that was extended to contiguous or has unbacked symints: https://github.com/pytorch/pytorch/pull/110520/files#diff-cf6ca00beddc32a2a6a2933fb9913b6a2b925ffc3b745488967210e4343134acR1735-R1739 bc `dynamic_reshape_indexer` fails with unbacked. The current api call (`want_contiguous`) does not enforce that it is contiguous, which I'm fixing in this pr. 

I agree, in the future, we should add support for `dynamic_reshape_indexer` & unbacked symints. I will file an issue for that.